core/domain/exp_jobs_one_off_test.py

```diff
@@ -239,6 +240,47 @@ def _run_batch_job_once_and_verify_output(
                         getattr(expected_job_output[exp_id], prop))
 
 
+class OneOffExplorationFirstPublishedJobTest(test_utils.GenericTestBase):
+
+    EXP_ID = 'exp_id'
+
+    def setUp(self):
+        super(OneOffExplorationFirstPublishedJobTest, self).setUp()
+
+    def test_first_published_time_of_exploration_that_is_unpublished(self):
+        """This tests that, if an exploration is published, unpublished, and
+        then published again, the job uses the first publication date as the
+        first published datetime.
+        """
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.login(self.ADMIN_EMAIL)
+        self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+        self.set_admins([self.ADMIN_EMAIL])
+
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.login(self.OWNER_EMAIL)
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+
+        exploration = self.save_new_valid_exploration(
+            self.EXP_ID, self.owner_id, end_state_name='End')
+        rights_manager.publish_exploration(self.owner_id, self.EXP_ID)
+        job_id = (exp_jobs_one_off.ExplorationFirstPublishedOneOffJob.create_new())
+        exp_jobs_one_off.ExplorationFirstPublishedOneOffJob.enqueue(job_id)
+        self.process_and_flush_pending_tasks()
+        exploration_rights = rights_manager.get_exploration_rights(self.EXP_ID)
+        exp_first_published = exploration_rights.first_published_in_msec
```

Comment:
I think here you should be checking equality of exploration_rights.first_published_msec and the exp_rights_model.last_updated, similar to what you did before.